client/tests/integration/smartcontracts/validator_with_migration_fail/src/lib.rs

```diff
@@ -10,131 +10,17 @@ extern crate panic_halt;
 use alloc::{borrow::ToOwned as _, format};
 
 use anyhow::anyhow;
-use iroha_validator::{
-    data_model::{
-        evaluate::{EvaluationError, ExpressionEvaluator},
-        ValidationFail,
-    },
-    iroha_wasm, parse,
-    prelude::*,
-};
-
-struct Validator {
-    verdict: Result,
-    block_height: u64,
-    host: iroha_wasm::Host,
-}
-
-impl Validator {
-    /// Construct [`Self`]
-    pub fn new(block_height: u64) -> Self {
-        Self {
-            verdict: Ok(()),
-            block_height,
-            host: iroha_wasm::Host,
+use iroha_validator::{data_model::ValidationFail, parse, prelude::*};
+
+default_validator!(
+    impl Visit for Validator {
+        fn visit_query(&mut self, _authority: &AccountId, _query: &QueryBox) {
+            self.deny(ValidationFail::NotPermitted(
+                "All queries are forbidden".to_owned(),
+            ));
         }
     }
```

Comment:
```suggestion
default_validator!(
    pub struct Validator {
        data1: u32,    // Just an example of additional fields in the validator
    }

    impl Visit for Validator {
        fn visit_query(&mut self, _authority: &AccountId, _query: &QueryBox) {
            self.deny(ValidationFail::NotPermitted(
                "All queries are forbidden".to_owned(),
            ));
        }
    }
```
You should not implicitly define a custom validator especially since you're referring to it even outside of the macro. Except for being able to control the name of you validator struct, you should also be able to define additional fields in the struct (such as `data1` in my example)